cdk/src/open-data-platform/data-plane/data-import/handler-factory.ts

```diff
@@ -1,22 +1,25 @@
 import { APIGatewayProxyResult } from 'aws-lambda';
 import * as AWS from 'aws-sdk';
+import { ParallelSink } from './utils/parallel-sink';
+import { offset, limit } from './utils/stream-utils';
 
 // These libraries don't have types, so they are imported in a different way.
 const { chain } = require('stream-chain');
 const { streamArray } = require('stream-json/streamers/StreamArray');
 const Batch = require('stream-json/utils/Batch');
 const Pick = require('stream-json/filters/Pick');
 
+const SINK_CONCURRENCY = 5;
```

Comment:
Does this write all water systems in one go? We previously had some timeout issues so I'm curious as to whether limiting to 5 at once would slow this down and cause OOM with large files.

Reply:
It's not writing them all in one go... it's limiting the number of batch writes that are in-flight at any moment to 5. 

Each 'batch' is still a configurable size (default 10 rows, in the case of water-systems I overrode it to 1k). 

The way streams work it *shouldn't* OOM because it should actually slow down the download to happen at the same speed as we're writing to the DB. 

That said, if you have a really large file in mind that I could test with I'd happily give it a go.